packages/ui/src/accounts/modals/ClaimVestingModal/ClaimVestingModal.tsx

```diff
@@ -0,0 +1,124 @@
+import { useMachine } from '@xstate/react'
+import React, { useEffect, useMemo, useState } from 'react'
+import styled from 'styled-components'
+
+import { ListHeader } from '@/accounts/components/Accounts'
+import { useMyAccounts } from '@/accounts/hooks/useMyAccounts'
+import { useMyBalances } from '@/accounts/hooks/useMyBalances'
+import { useTransactionFee } from '@/accounts/hooks/useTransactionFee'
+import { useVesting } from '@/accounts/hooks/useVesting'
+import { SelectVestingAccount } from '@/accounts/modals/ClaimVestingModal/components/SelectVestingAccount'
+import { Account } from '@/accounts/types'
+import { FailureModal } from '@/common/components/FailureModal'
+import { InputComponent } from '@/common/components/forms'
+import { Modal, ModalBody, ModalHeader, ModalTransactionFooter } from '@/common/components/Modal'
+import { RowGapBlock } from '@/common/components/page/PageContent'
+import { SuccessModal } from '@/common/components/SuccessModal'
+import { TextMedium, TokenValue } from '@/common/components/typography'
+import { useApi } from '@/common/hooks/useApi'
+import { useModal } from '@/common/hooks/useModal'
+import { useSignAndSendTransaction } from '@/common/hooks/useSignAndSendTransaction'
+import { transactionMachine } from '@/common/model/machines'
+
+export const ClaimVestingModal = () => {
+  const { hideModal } = useModal()
+  const { api } = useApi()
+  const [selectedAccount, setSelectedAccount] = useState<Account>()
+  const vesting = useVesting(selectedAccount?.address)
+  const balances = useMyBalances()
+  const { allAccounts } = useMyAccounts()
+  const [state, , service] = useMachine(transactionMachine)
+
+  useTransactionFee(selectedAccount?.address, () => api?.tx.vesting.vest())
+
+  useEffect(() => {
+    if (balances) {
+      Object.entries(balances).forEach(([key, value]) => {
+        if (value.isVesting) {
+          setSelectedAccount(allAccounts.find((a) => a.address === key))
+        }
+      })
+    }
+  }, [!balances])
+
+  const transaction = useMemo(() => api?.tx.vesting.vest(), [])
```

Comment:
This would work:
```suggestion
  const { transaction } = useTransactionFee(selectedAccount?.address, () => api?.tx.vesting.vest())
```

Although ideally in the future we should probably refactor `useSignAndSendTransaction` to accept a callback and update the `TransactionFeesContext`

Reply:
If we take transaction out of `useMemo` it will cause loop cause `useSignAndSendTransaction` will receive new object on every re-render. 

Maybe we can stick to this? WDYT
```
  const transaction = useMemo(() => api?.tx.vesting.vest(), [])

  useTransactionFee(selectedAccount?.address, () => transaction, [transaction])

  const { isReady, sign } = useSignAndSendTransaction({
    transaction,
    signer: selectedAccount?.address ?? '',
    service: service as any,
  })
  ```